src/Data.cpp

```diff
@@ -0,0 +1,609 @@
+/// @param[in] left_flexbounds_name
+/// The name of the left flexbounds, which is a 2-tuple of read/MSA positions
+/// providing the bounds of the germline's left flex region.
+/// @param[in] right_flexbounds_name
+/// The name of the right flexbounds, which is a 2-tuple of read/MSA positions
+/// providing the bounds of the germline's right flex region.
+/// @return
+/// A germline match probability matrix (filled only with transition
+/// probabilities).
+///
+/// This function computes the probabilities of germline transition paths
+/// starting in the left flexbounds and ending in the right flexbounds.  We
+/// account for left/right flexbounds positions without germline states by
+/// assigning germline paths through these positions a probability of zero.
+Eigen::MatrixXd Data::GermlineTransProbMatrix(
+    GermlinePtr germ_ptr, std::string left_flexbounds_name,
+    std::string right_flexbounds_name) const {
+  // Extract the left/right flexbounds, match indices, and relpos.
+  std::pair<int, int> left_flexbounds = flexbounds_.at(left_flexbounds_name);
+  std::pair<int, int> right_flexbounds = flexbounds_.at(right_flexbounds_name);
+  std::array<int, 6> match_indices =
+      match_indices_.at({germ_ptr->name(), left_flexbounds_name});
+  int match_start = match_indices[kMatchStart];
+  int match_end = match_indices[kMatchEnd];
+  int left_flex = match_indices[kLeftFlex];
+  int right_flex = match_indices[kRightFlex];
+  int row_start = match_indices[kRowStart];
+  int col_start = match_indices[kColStart];
+  int relpos = relpos_.at(germ_ptr->name());
+
+  assert(left_flexbounds.first <= left_flexbounds.second);
+  assert(right_flexbounds.first <= right_flexbounds.second);
+  assert(left_flexbounds.first + 1 <= right_flexbounds.first);
+  assert(left_flexbounds.second + 1 <= right_flexbounds.second);
+
+  assert(relpos < right_flexbounds.second);
+  assert(right_flexbounds.first <= relpos + germ_ptr->length());
+
+  assert(0 <= left_flexbounds.first);
+  assert(right_flexbounds.second <= this->length());
+
+  Eigen::MatrixXd match_matrix = Eigen::MatrixXd::Zero(
+      left_flexbounds.second - left_flexbounds.first + 1,
+      right_flexbounds.second - right_flexbounds.first + 1);
+
+  // If the germline's left flex region has no germline states,
+  // return a match probability matrix filled only with zeroes.
+  if (relpos > left_flexbounds.second) return match_matrix;
+
+  // Compute the transition probability portion of the germline match matrix.
+  match_matrix.block(row_start, col_start, left_flex + 1, right_flex + 1) =
+      germ_ptr->transition()
+          .block(match_start - relpos, match_start - relpos,
+                 match_end - match_start, match_end - match_start)
+          .block(0, match_end - match_start - right_flex - 1, left_flex + 1,
+                 right_flex + 1);
+
+  return match_matrix;
+};
+
+
+/// @brief Creates the path probability matrix for the non-templated insertion
```

Comment:
Please make web link to issue explaining the logic.

Reply:
See https://github.com/matsengrp/linearham/issues/3#issuecomment-256169007 for NTI smooshable calculation.